### PR TITLE
Policy Updates for new version

### PR DESCRIPTION
**Pull request by pratimaupadhyay02**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by ravbhatnagar** [comment]
@pratimaupadhyay02  removing APIs from stable api-version is a breaking change. The Azure API deprecation process must be followed before APIs can be deprecated.

**Comment by pratimaupadhyay02** [comment]
> removing APIs from stable api-version is a breaking change. The Azure API deprecation process must be followed before APIs can be deprecated.

We are not removing this api version completely. We are just upgrading the api version with this SDK release. This is not deprecation.

**Comment by ravbhatnagar** [comment]
not sure what the issue is. Is the GitHub diff messed up? It is showing bunch of APIs in red.

**Comment by TAdityaAnirudh** [comment]
> 
> 
> not sure what the issue is. Is the GitHub diff messed up? It is showing bunch of APIs in red.

Hey Gaurav. We are a little lost here. Are you saying we cannot move APIs across version from one SDK version to another? Just to reiterate, we are not removing these APIs. We are just bumping up their version. We had done this just a month back for more than 20 APIs. So, what's the issue you're talking about? Are we missing something here?

**Comment by erich-wang** [comment]
@ravbhatnagar, could you please respond?

**Comment by ravbhatnagar** [comment]
@pratimaupadhyay02 @TAdityaAnirudh - you can add apis to a newer api-version. But that does not mean you can remove apis from an existing api-version. If you want these apis to be supported in a newer api-version, create a new api-version and add them there. They must not be removed from existing GA api-version, which I am assuming these are. Does that make sense?

**Comment by pratimaupadhyay02** [comment]
> @pratimaupadhyay02 @TAdityaAnirudh - you can add apis to a newer api-version. But that does not mean you can remove apis from an existing api-version. If you want these apis to be supported in a newer api-version, create a new api-version and add them there. They must not be removed from existing GA api-version, which I am assuming these are. Does that make sense?

@ravbhatnagar We tried retaining the APIs in the existing GA-api version. Compilation of specs fails in that case. So it's not an option.
The changes in this PR are very similar to another PR we checked in around a month back.
https://github.com/Azure/azure-rest-api-specs/pull/7345/files
Has anything changed within this one month?
The older version is still active and is still supported. Any customer who uses an older version of Nuget can use that API version. We are not deprecating an API version with this. We just want our customers to use the latest version for these APIs. So, we are moving these around. 
We can setup a call and discuss this if required.

**Comment by ravbhatnagar** [comment]
please work with @sanjaiganesh as he is the current API review oncall.

**Comment by KrisBash** [comment]
@pratimaupadhyay02 please do not remove active APIs from prior versions. Users must be able to use all supported API versions in latest SDK builds

**Comment by erich-wang** [comment]
@KrisBash , please help to review again, thanks.

File: specification/recoveryservicesbackup/resource-manager/readme.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ These settings apply only when `--tag=package-2019-05` is specified on the comma
 
 ``` yaml $(tag) == 'package-2019-05'
 input-file:
-- Microsoft.RecoveryServices/stable/2019-05-13/bms.json
+- Microsoft.RecoveryServices/stable/2019-06-15/bms.json
```
Comment (anthony-c-martin) [inline specification/recoveryservicesbackup/resource-manager/readme.md]: Please keep the old entry (2019-05-13) in-place

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]
Please could you describe the changes you're making with this PR? When I diff bms.json against the previous version (2019-05-13), I see virtually no changes - is that intentional?

```diff
diff --git "a/2019-05-13\\bms.json" "b/2019-06-15\\bms.json"
index 24274a9..a77e080 100644
--- "a/2019-05-13\\bms.json"
+++ "b/2019-06-15\\bms.json"
@@ -1,7 +1,7 @@
 {
   "swagger": "2.0",
   "info": {
-    "version": "2019-05-13",
+    "version": "2019-06-15",
     "title": "RecoveryServicesBackupClient",
     "x-ms-code-generation-settings": {
       "internalConstructors": false
@@ -3141,6 +3141,17 @@
       },
       "x-ms-discriminator-value": "GenericProtectionPolicy"
     },
+    "InstantRPAdditionalDetails": {
+      "type": "object",
+      "properties": {
+        "azureBackupRGNamePrefix": {
+          "type": "string"
+        },
+        "azureBackupRGNameSuffix": {
+          "type": "string"
+        }
+      }
+    },
     "GenericProtectedItem": {
       "description": "Base class for backup items.",
       "type": "object",
@@ -4080,6 +4091,9 @@
     },
     "ProtectedItem": {
       "description": "Base class for backup items.",
+      "required": [
+        "protectedItemType"
+      ],
       "type": "object",
       "properties": {
         "protectedItemType": {
```

File: specification/recoveryservicesbackup/resource-manager/readme.md (hunk 1 above)
Comment (pratimaupadhyay02) [reply specification/recoveryservicesbackup/resource-manager/readme.md]: Retained the old entry and created a new package (2019-06) which contains all the APIs.

**Comment by pratimaupadhyay02** [comment]
> Please could you describe the changes you're making with this PR? When I diff bms.json against the previous version (2019-05-13), I see virtually no changes - is that intentional?
> diff --git "a/2019-05-13\\bms.json" "b/2019-06-15\\bms.json"
> index 24274a9..a77e080 100644
> --- "a/2019-05-13\\bms.json"
> +++ "b/2019-06-15\\bms.json"
> @@ -1,7 +1,7 @@
>  {
>    "swagger": "2.0",
>    "info": {
> -    "version": "2019-05-13",
> +    "version": "2019-06-15",
>      "title": "RecoveryServicesBackupClient",
>      "x-ms-code-generation-settings": {
>        "internalConstructors": false
> @@ -3141,6 +3141,17 @@
>        },
>        "x-ms-discriminator-value": "GenericProtectionPolicy"
>      },
> +    "InstantRPAdditionalDetails": {
> +      "type": "object",
> +      "properties": {
> +        "azureBackupRGNamePrefix": {
> +          "type": "string"
> +        },
> +        "azureBackupRGNameSuffix": {
> +          "type": "string"
> +        }
> +      }
> +    },
>      "GenericProtectedItem": {
>        "description": "Base class for backup items.",
>        "type": "object",
> @@ -4080,6 +4091,9 @@
>      },
>      "ProtectedItem": {
>        "description": "Base class for backup items.",
> +      "required": [
> +        "protectedItemType"
> +      ],
>        "type": "object",
>        "properties": {
>          "protectedItemType": {

Yeah it is intentional. We are just adding the property RGNamePrefix and RGNameSuffix in AzureIaasVmProtectionPolicy and this contains the changes related to it only. Other than that we are bumping up the version of all the APIs to 2019-06-15.

**Comment by anthony-c-martin** [comment]
> Yeah it is intentional. We are just adding the property RGNamePrefix and RGNameSuffix in AzureIaasVmProtectionPolicy and this contains the changes related to it only. Other than that we are bumping up the version of all the APIs to 2019-06-15.

Thanks for the explanation! I see that `InstantRPAdditionalDetails` is being added as a definition but isn't being referenced anywhere (so I don't think it's going to have any effect on the shape of the API). Should it be referenced as a property on the `AzureIaasVmProtectionPolicy` definition?

**Review by anthony-c-martin — APPROVED** [review]
LGTM, thanks!

**Pull request merged**